### PR TITLE
dfu with sidewalk fuota

### PR DESCRIPTION
**Pull request by ktaborowski**
## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: 

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [x] Change has been tested.
- [x] Tests were updated (if applicable).

### CODE REVIEW

File: samples/sid_end_device/src/sidewalk.c [hunk 1]
```diff
@@ -316,31 +317,26 @@ static void state_sidewalk_run(void *o)
...
-		if (e != SID_ERROR_NONE) {
-			LOG_ERR("Failed to hash received file transfer with error %s",
-				SID_ERROR_T_STR(e));
-		} else {
-#define HEX_PRINTER(a, ...) "%02X"
-#define HEX_PRINTER_ARG(a, ...) hash_out[a]
-			char hex_str[sizeof(hash_out) * 2 + 1] = { 0 };
-			snprintf(hex_str, sizeof(hex_str), LISTIFY(32, HEX_PRINTER, ()),
-				 LISTIFY(32, HEX_PRINTER_ARG, (, )));
-			LOG_INF("SHA256: %s", hex_str);
```
Comment (RobertGalatNordic) [inline samples/sid_end_device/src/sidewalk.c:343]: Our CI tests rely on the hash output of the payload so we shouldn't delete this (at liest not without changes made in tests)

 would propose to extract this to a function (to help with swapping this implementation to new one) and contitionally call the extracted function ( or the new implementation) based on the KConfig feature flag setting. 
 
 I would propose to set the default value of this feature flag to use the old implementation.
 
 Not everything need to be behind the feature flag setting. The changes in the structures, additional fields does not impact previous implementation, so they can be added as is.

Comment (ktaborowski) [reply samples/sid_end_device/src/sidewalk.c:343]: thanks for pointing this out, will be reversed before converting draft to pr

File: samples/sid_end_device/src/file_transfer.c [hunk 2]
```diff
@@ -159,6 +201,54 @@ void app_file_transfer_demo_init(struct sid_handle *handle)
...
+					   is_last);
+	if (err) {
+		LOG_ERR("Flash image write err %d", err);
+		LOG_INF("Cancelig file transfer");
+		ret = sid_bulk_data_transfer_cancel(
+			handle, transfer->file_id, SID_BULK_DATA_TRANSFER_REJECT_REASON_FILE_TOO_BIG);
+		if (ret != SID_ERROR_NONE) {
+			LOG_ERR("sbdt cancel ret %s", SID_ERROR_T_STR(ret));
+		}
+	} else {
```
Comment (RobertGalatNordic) [inline samples/sid_end_device/src/file_transfer.c:239]: Just a cosmetic suggestion, this else is not needed

**Comment by RobertGalatNordic** [comment]
/__w/sdk-sidewalk/sdk-sidewalk/sidewalk/samples/sid_end_device/src/sidewalk.c:180:13: error: 'print_hash' defined but not used [-Werror=unused-function]

File: samples/sid_end_device/src/sidewalk.c [hunk 3]
```diff
@@ -175,6 +177,28 @@ static void state_sidewalk_entry(void *o)
 #endif /* CONFIG_SIDEWALK_AUTO_START */
 }
 
+static void print_hash(void *data, size_t data_size)
```
Comment (RobertGalatNordic) [inline samples/sid_end_device/src/sidewalk.c:180]: unused when Kconfig CONFIG_SIDEWALK_FILE_TRANSFER disabled

**Review by RobertGalatNordic — APPROVED** [review]

**Review by totyz — APPROVED** [review]
Tests PASSED

File: doc/samples/sid_end_device.rst [hunk 4]
```diff
@@ -129,6 +129,32 @@ For example:
...
     However, if the build type is not selected, the ``debug`` build type is used by default.
 
+Device Firmware Upgrade support
+===============================
+
+The sample supports over-the-air (OTA) device firmware upgrade (DFU) using one of the two following protocols:
+
+* Simple Management Protocol (SMP) over Bluetooth® LE.
+   In this case, the DFU can be done either using a smartphone application or a PC command line tool.
+   Note that this protocol is not part of the Sidewalk specification.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:139]:
```suggestion
  In this case, the DFU is performed using a smartphone application or a PC command line tool.
   This protocol is not a part of the Sidewalk specification.
```

File: doc/samples/sid_end_device.rst [hunk 5]
```diff
@@ -129,6 +129,32 @@ For example:
...
+* Downlink file transfer over Bluetooth LE (experimental).
+   The Sidewalk Bulk Data Transfer mode and Integration of AWS IoT FUOTA service allow sending files to fleet of IoT devices from the AWS IoT FUOTA task.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:141]:
```suggestion
  This method uses the Sidewalk Bulk Data Transfer (SBDT) mode and integration of AWS IoT FUOTA service to send files to fleet of IoT devices from the AWS IoT FUOTA task.
```

File: doc/samples/sid_end_device.rst [hunk 6]
```diff
@@ -129,6 +129,32 @@ For example:
...
+   For more details, see Sidewalk Bulk Data Transfer section in the `Amazon Sidewalk specification`_.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:142]:
```suggestion
   For more details, see the Sidewalk Bulk Data Transfer section in the `Amazon Sidewalk specification`_.
```

File: doc/samples/sid_end_device.rst [hunk 7]
```diff
@@ -129,6 +129,32 @@ For example:
...
+
+   .. note::
+
+      To test this functionality, it is required to have access to S3 bucket.
+      If you want to gain the access, contact your account manager or sidewalk-beta-support@amazon.com.
+      In the message provide your AWS account ID to be added to the allowlist.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:148]:
```suggestion
      To test this functionality, you must have access to S3 bucket.
      To gain it, contact your account manager or sidewalk-beta-support@amazon.com.
      In the message provide your AWS account ID to be added to the allowlist.
```

File: doc/samples/sid_end_device.rst [hunk 8]
```diff
@@ -129,6 +129,32 @@ For example:
...
+
+In both cases, MCUboot secure bootloader is used to apply the new firmware image.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:150]:
```suggestion
In both cases, the MCUboot secure bootloader is used to apply the new firmware image.
```

File: doc/samples/sid_end_device.rst [hunk 9]
```diff
@@ -129,6 +129,32 @@ For example:
...
+
+The DFU over SMP is enabled by default.
+
+The following configuration arguments are available during the build process for configuring DFU:
+
+* To configure the sample to support the downlink file transfer over Bluetooth LE, use the ``-DCONFIG_SIDEWALK_FILE_TRANSFER=y`` build flag.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:156]:
```suggestion
* To enable support for the downlink file transfer over Bluetooth LE, use the ``-DCONFIG_SIDEWALK_FILE_TRANSFER=y`` build flag.
```

File: doc/samples/sid_end_device.rst [hunk 10]
```diff
@@ -135,25 +135,25 @@ Device Firmware Upgrade support
 The sample supports over-the-air (OTA) device firmware upgrade (DFU) using one of the two following protocols:
 
 * Simple Management Protocol (SMP) over Bluetooth® LE.
-   In this case, the DFU can be done either using a smartphone application or a PC command line tool.
-   Note that this protocol is not part of the Sidewalk specification.
+  In this case, the DFU is performed using a smartphone application or a PC command line tool.
+   This protocol is not a part of the Sidewalk specification.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:139]:
```suggestion
  This protocol is not a part of the Sidewalk specification.
```
sorry - forgot to align last sentences of these 2 bullet points

File: doc/samples/sid_end_device.rst [hunk 11]
```diff
@@ -135,25 +135,25 @@ Device Firmware Upgrade support
...
 * Simple Management Protocol (SMP) over Bluetooth® LE.
-   In this case, the DFU can be done either using a smartphone application or a PC command line tool.
-   Note that this protocol is not part of the Sidewalk specification.
+  In this case, the DFU is performed using a smartphone application or a PC command line tool.
+   This protocol is not a part of the Sidewalk specification.
 * Downlink file transfer over Bluetooth LE (experimental).
-   The Sidewalk Bulk Data Transfer mode and Integration of AWS IoT FUOTA service allow sending files to fleet of IoT devices from the AWS IoT FUOTA task.
-   For more details, see Sidewalk Bulk Data Transfer section in the `Amazon Sidewalk specification`_.
+  This method uses the Sidewalk Bulk Data Transfer (SBDT) mode and integration of AWS IoT FUOTA service to send files to fleet of IoT devices from the AWS IoT FUOTA task.
+   For more details, see the Sidewalk Bulk Data Transfer section in the `Amazon Sidewalk specification`_.
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:142]:
```suggestion
  For more details, see the Sidewalk Bulk Data Transfer section in the `Amazon Sidewalk specification`_.
```

File: doc/includes/memory_requirement.txt [hunk 12]
```diff
@@ -3,50 +3,50 @@
...
+      |                :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Debug)                 |      26.33       |        512.94        |         4         |            8             |     551.27     |     122.16     |
       +----------------------------------------------------------------------------------------------------------------+------------------+----------------------+-------------------+--------------------------+----------------+----------------+
-      |  :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Release)                             |      26.33       |        430.45        |         4         |            8             |     468.78     |     105.04     |
+      |               :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Release)                |      26.33       |        430.49        |         4         |            8             |     468.82     |     106.04     |
```
Comment (annwoj) [inline doc/includes/memory_requirement.txt:17]: can these empty spaces be aligned as per the previous version?

File: doc/includes/memory_requirement.txt [hunk 13]
```diff
@@ -3,50 +3,50 @@
...
       +----------------------------------------------------------------------------------------------------------------+------------------+----------------------+-------------------+--------------------------+----------------+----------------+
 
    .. tab:: nRF5340 DK
 
       The following table lists the total memory requirements for the default variant of the :ref:`Sidewalk_End_device` sample running on the `nRF5340 DK`_.
-      To determine the available space for user extension, it needs to be summarized against the allocated partition size and its layout.
+      To determine the available space for user extension, it needs to be summarized versus the allocated partition size and its layout.
```
Comment (annwoj) [inline doc/includes/memory_requirement.txt:23]:
```suggestion
      To determine the available space for user extension, it needs to be summarized against the allocated partition size and its layout.
```
as per previous version - in technical documentation we avoid using latin words

File: doc/includes/memory_requirement.txt [hunk 14]
```diff
@@ -3,50 +3,50 @@
...
+      |  :ref:`Hello Sidewalk Bluetooth LE <variant_sidewalk_hello>` (``CONFIG_SIDEWALK_SUBGHZ_SUPPORT=n``, Release)   |      30.77       |        291.82        |         4         |            8             |     334.59     |     79.40      |
       +----------------------------------------------------------------------------------------------------------------+------------------+----------------------+-------------------+--------------------------+----------------+----------------+
-      |  :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Debug)                               |      30.77       |        435.18        |         4         |            8             |     477.95     |     98.34      |
+      |                :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Debug)                 |      30.77       |        452.05        |         4         |            8             |     494.82     |     110.62     |
```
Comment (annwoj) [inline doc/includes/memory_requirement.txt:32]: as above

File: doc/includes/memory_requirement.txt [hunk 15]
```diff
@@ -3,50 +3,50 @@
...
+      |               :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Release)                |      30.77       |        367.28        |         4         |            8             |     410.05     |     94.58      |
       +----------------------------------------------------------------------------------------------------------------+------------------+----------------------+-------------------+--------------------------+----------------+----------------+
 
    .. tab:: nRF54L15 PDK
 
       The following table lists the total memory requirements for the default variant of the :ref:`Sidewalk_End_device` sample running on the `nRF54L15 PDK`_.
-      To determine the available space for user extension, it needs to be summarized against the allocated partition size and its layout.
+      To determine the available space for user extension, it needs to be summarized versus the allocated partition size and its layout.
```
Comment (annwoj) [inline doc/includes/memory_requirement.txt:40]:
```suggestion
      To determine the available space for user extension, it needs to be summarized against the allocated partition size and its layout.
```

File: doc/includes/memory_requirement.txt [hunk 16]
```diff
@@ -3,50 +3,50 @@
...
+      |  :ref:`Hello Sidewalk Bluetooth LE <variant_sidewalk_hello>` (``CONFIG_SIDEWALK_SUBGHZ_SUPPORT=n``, Release)   |      26.15       |        285.78        |         4         |            8             |     323.93     |     84.77      |
       +----------------------------------------------------------------------------------------------------------------+------------------+----------------------+-------------------+--------------------------+----------------+----------------+
-      |  :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Debug)                               |      26.03       |        432.32        |         4         |            8             |     470.35     |     103.99     |
+      |                :ref:`Hello Sidewalk Bluetooth LE and sub-GHz <variant_sidewalk_hello>` (Debug)                 |      26.15       |        449.00        |         4         |            8             |     487.15     |     116.28     |
```
Comment (annwoj) [inline doc/includes/memory_requirement.txt:49]: as above

File: doc/includes/memory_requirement.txt [hunk 17]
```diff
@@ -3,50 +3,50 @@
    .. tab:: nRF52840 DK
 
       The following table lists the total memory requirements for the default variant of the :ref:`Sidewalk_End_device` sample running on the `nRF52840 DK`_.
-      To determine the available space for user extension, it needs to be summarized against the allocated partition size and its layout.
+      To determine the available space for user extension, it needs to be summarized versus the allocated partition size and its layout.
```
Comment (annwoj) [inline doc/includes/memory_requirement.txt:6]:
```suggestion
      To determine the available space for user extension, it needs to be summarized against the allocated partition size and its layout.
```

**Review by annwoj — APPROVED** [review]

**Pull request merged**